### PR TITLE
Panel translation fragments

### PR DESCRIPTION
**Pull request by emlove**
## Description
This PR adds translations for the shopping list panel as a proof of concept for our panel-based translation fragmenting. Similar to our Polymer files, we want to split our translation resources so that translations only used on individual panels don't need to be part of the core translation set the client downloads immediately.

**Backend PR:** https://github.com/home-assistant/home-assistant/pull/10937

### CODE REVIEW

File: panels/shopping-list/ha-panel-shopping-list.html [hunk 1]
```diff
@@ -79,7 +80,7 @@
             <paper-listbox slot="dropdown-content">
               <paper-item
                 on-tap="_clearCompleted"
-              >Clear completed</paper-item>
+              >[[haLocalize('ui.panel.shopping-list', 'clear_completed')]]</paper-item>
```
Comment (balloob) [inline panels/shopping-list/ha-panel-shopping-list.html]: Does it make sense to prefix it with ui? Doesn't panel already imply that?

File: script/build_frontend [hunk 2]
```diff
@@ -60,6 +59,7 @@ gzip -f -n -k -9 \
     *.js \
     ./panels/*.html \
     ./translations/*.json \
+    ./translations/*/*.json \
```
Comment (balloob) [inline script/build_frontend]: When working on #678 I realized that `script/build_frontend` shouldn't exist. It's like doing all these tasks that we should turn into gulp tasks. Gzipping a folder is easily done in gulp: [example](https://github.com/home-assistant/home-assistant-polymer/pull/678/files#diff-0f4211b4a498fe0ef4990a1a55e17ec1R38)

File: src/layouts/partial-panel-resolver.html [hunk 3]
```diff
@@ -74,6 +75,8 @@
         type: Object,
         computed: 'computeCurrentPanel(hass, routeData)',
         observer: 'panelChanged',
+        notify: true,
```
Comment (balloob) [inline src/layouts/partial-panel-resolver.html]: I don't think that it makes sense to compute this here and then propagate it all up? It makes more sense to compute this in `<home-assistant>` and pass it down as a property on `hass`

File: src/translations/en.json [hunk 4]
```diff
@@ -216,5 +216,14 @@
       "home": "[%key:state::device_tracker::home%]",
       "not_home": "[%key:state::device_tracker::not_home%]"
     }
+  },
+  "ui": {
+    "panel": {
+      "shopping-list": {
+        "clear_completed": "Clear completed",
+        "add_item": "Add item",
+        "microphone_tip": "Tap the microphone on the top right and say “Add candy to my shopping list”"
```
Comment (balloob) [inline src/translations/en.json]: Our voice UI is currently only working in English, so is it possible we can add a note to this translation key to not translate the text within quotes?

File: panels/shopping-list/ha-panel-shopping-list.html (hunk 1 above)
Comment (armills) [reply panels/shopping-list/ha-panel-shopping-list.html]: What I was envisioning is `ui` would be a top-level container for all frontend text that is simply used for display purposes, and isn't semantically linked to anything on the backend. The panel namespace here would contain the text we plan to fragment off into different files, but there would also likely be `ui` text that belongs in the core data set if it's used in more-info displays, etc. This was partially also driven by the fact that `panel` is already a top level key containing our panel names.

That being said this was just my first attempt at this. Open to suggestions or other proposals here.

File: script/build_frontend (hunk 2 above)
Comment (armills) [reply script/build_frontend]: Agree with that, but I'd suggest we migrate this in a separate PR.

Comment (balloob) [reply script/build_frontend]: Agree

File: panels/shopping-list/ha-panel-shopping-list.html (hunk 1 above)
Comment (balloob) [reply panels/shopping-list/ha-panel-shopping-list.html]: Makes sense.

**Review by balloob — APPROVED** [review]

**Comment by armills** [comment]
Still have it on my list to try and move panel calcs to the top-level. Just haven't had much time the past few days and it's going to take a bit of legwork to reorganize the app-location and app-route usage.

**Comment by armills** [comment]
I went for an approach where we calculate the panelUrl at the top-level, and pass it down through the normal polymer property methods. It feels a little overkill to attach it to the hass object since it's not often used, although we can always do that instead.

**Comment by balloob** [comment]
attaching something to the hass object is cheap if it is not updated often.

**Comment by balloob** [comment]
Also, everything else is attached to the hass object.

**Comment by armills** [comment]
OK, no problem. I'll put it on hass.

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
This is ok to merge. Is it ready to merge?

**Comment by armills** [comment]
👍 Yep

**Pull request merged**